### PR TITLE
 Systemtap: 4.1 -> 4.5 

### PR DESCRIPTION
**Pull request by KaiHa**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/profiling/systemtap/default.nix [hunk 1]
```diff
@@ -16,6 +16,7 @@ let
     pname = "systemtap";
     inherit version;
     src = fetchgit { inherit url rev sha256; };
+    patches = [./PR27001.patch];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/profiling/systemtap/default.nix:19]:
```suggestion
    patches = [ ./PR27001.patch ];
```

**Comment by karolchmist** [comment]
Result of `nixpkgs-review pr 114691` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.bcc</li>
    <li>linuxPackages-libre.bpftrace</li>
    <li>linuxPackages-libre.oci-seccomp-bpf-hook</li>
    <li>linuxPackages-libre.systemtap</li>
    <li>linuxPackages_hardkernel_4_14.bcc</li>
    <li>linuxPackages_hardkernel_4_14.bpftrace</li>
    <li>linuxPackages_hardkernel_4_14.systemtap</li>
    <li>linuxPackages_hardkernel_latest.bcc</li>
    <li>linuxPackages_hardkernel_latest.bpftrace</li>
    <li>linuxPackages_hardkernel_latest.systemtap</li>
    <li>linuxPackages_latest-libre.bcc</li>
    <li>linuxPackages_latest-libre.bpftrace</li>
    <li>linuxPackages_latest-libre.oci-seccomp-bpf-hook</li>
    <li>linuxPackages_latest-libre.systemtap</li>
  </ul>
</details>
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_4_14.perf</li>
    <li>linuxPackages_4_19.perf</li>
    <li>linuxPackages_4_4.perf</li>
    <li>linuxPackages_4_9.perf</li>
    <li>linuxPackages_hardkernel_4_14.perf (linuxPackages_hardkernel_latest.perf)</li>
  </ul>
</details>
<details>
  <summary>53 packages built:</summary>
  <ul>
    <li>cargo-flamegraph</li>
    <li>linuxPackages-libre.perf</li>
    <li>linuxPackages.bcc (linuxPackages_5_10.bcc ,linuxPackages_xen_dom0.bcc)</li>
    <li>linuxPackages.bpftrace (linuxPackages_5_10.bpftrace ,linuxPackages_xen_dom0.bpftrace)</li>
    <li>linuxPackages.oci-seccomp-bpf-hook (linuxPackages_5_10.oci-seccomp-bpf-hook ,linuxPackages_xen_dom0.oci-seccomp-bpf-hook)</li>
    <li>linuxPackages.perf (linuxPackages_5_10.perf ,linuxPackages_hardened.perf ,linuxPackages_xen_dom0.perf ,linuxPackages_xen_dom0_hardened.perf)</li>
    <li>linuxPackages.systemtap (linuxPackages_5_10.systemtap ,linuxPackages_xen_dom0.systemtap)</li>
    <li>linuxPackages_4_14.bcc</li>
    <li>linuxPackages_4_14.bpftrace</li>
    <li>linuxPackages_4_14.systemtap</li>
    <li>linuxPackages_4_19.bcc</li>
    <li>linuxPackages_4_19.bpftrace</li>
    <li>linuxPackages_4_19.systemtap</li>
    <li>linuxPackages_4_4.bcc</li>
    <li>linuxPackages_4_4.bpftrace</li>
    <li>linuxPackages_4_4.systemtap</li>
    <li>linuxPackages_4_9.bcc</li>
    <li>linuxPackages_4_9.bpftrace</li>
    <li>linuxPackages_4_9.systemtap</li>
    <li>linuxPackages_5_11.bcc (linuxPackages_latest.bcc ,linuxPackages_latest_xen_dom0.bcc)</li>
    <li>linuxPackages_5_11.bpftrace (linuxPackages_latest.bpftrace ,linuxPackages_latest_xen_dom0.bpftrace)</li>
    <li>linuxPackages_5_11.oci-seccomp-bpf-hook (linuxPackages_latest.oci-seccomp-bpf-hook ,linuxPackages_latest_xen_dom0.oci-seccomp-bpf-hook)</li>
    <li>linuxPackages_5_11.perf (linuxPackages_latest.perf ,linuxPackages_latest_hardened.perf ,linuxPackages_latest_xen_dom0.perf ,linuxPackages_latest_xen_dom0_hardened.perf)</li>
    <li>linuxPackages_5_11.systemtap (linuxPackages_latest.systemtap ,linuxPackages_latest_xen_dom0.systemtap)</li>
    <li>linuxPackages_5_4.bcc</li>
    <li>linuxPackages_5_4.bpftrace</li>
    <li>linuxPackages_5_4.oci-seccomp-bpf-hook</li>
    <li>linuxPackages_5_4.perf</li>
    <li>linuxPackages_5_4.systemtap</li>
    <li>linuxPackages_hardened.bcc (linuxPackages_xen_dom0_hardened.bcc)</li>
    <li>linuxPackages_hardened.bpftrace (linuxPackages_xen_dom0_hardened.bpftrace)</li>
    <li>linuxPackages_hardened.oci-seccomp-bpf-hook (linuxPackages_xen_dom0_hardened.oci-seccomp-bpf-hook)</li>
    <li>linuxPackages_hardened.systemtap (linuxPackages_xen_dom0_hardened.systemtap)</li>
    <li>linuxPackages_latest-libre.perf</li>
    <li>linuxPackages_latest_hardened.bcc (linuxPackages_latest_xen_dom0_hardened.bcc)</li>
    <li>linuxPackages_latest_hardened.bpftrace (linuxPackages_latest_xen_dom0_hardened.bpftrace)</li>
    <li>linuxPackages_latest_hardened.oci-seccomp-bpf-hook (linuxPackages_latest_xen_dom0_hardened.oci-seccomp-bpf-hook)</li>
    <li>linuxPackages_latest_hardened.systemtap (linuxPackages_latest_xen_dom0_hardened.systemtap)</li>
    <li>linuxPackages_lqx.bcc</li>
    <li>linuxPackages_lqx.bpftrace</li>
    <li>linuxPackages_lqx.oci-seccomp-bpf-hook</li>
    <li>linuxPackages_lqx.perf</li>
    <li>linuxPackages_lqx.systemtap</li>
    <li>linuxPackages_testing_bcachefs.bcc</li>
    <li>linuxPackages_testing_bcachefs.bpftrace</li>
    <li>linuxPackages_testing_bcachefs.oci-seccomp-bpf-hook</li>
    <li>linuxPackages_testing_bcachefs.perf</li>
    <li>linuxPackages_testing_bcachefs.systemtap</li>
    <li>linuxPackages_zen.bcc</li>
    <li>linuxPackages_zen.bpftrace</li>
    <li>linuxPackages_zen.oci-seccomp-bpf-hook</li>
    <li>linuxPackages_zen.perf</li>
    <li>linuxPackages_zen.systemtap</li>
  </ul>
</details>

File: pkgs/development/tools/profiling/systemtap/default.nix [hunk 2]
```diff
@@ -16,6 +16,7 @@ let
     pname = "systemtap";
     inherit version;
     src = fetchgit { inherit url rev sha256; };
+    patches = [ ./PR27001.patch ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/profiling/systemtap/default.nix:19]: Please fetchpatch this.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 114691` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
20 packages marked as broken and skipped:
</summary>
<ul>
<li>linuxPackages-libre.bcc</li>
<li>linuxPackages-libre.bpftrace</li>
<li>linuxPackages-libre.oci-seccomp-bpf-hook</li>
<li>linuxPackages-libre.systemtap</li>
<li>linuxPackages_4_14.perf</li>
<li>linuxPackages_4_19.perf</li>
<li>linuxPackages_4_4.perf</li>
<li>linuxPackages_4_9.perf</li>
<li>linuxPackages_hardkernel_4_14.bcc</li>
<li>linuxPackages_hardkernel_4_14.bpftrace</li>
<li>linuxPackages_hardkernel_4_14.perf</li>
<li>linuxPackages_hardkernel_4_14.systemtap</li>
<li>linuxPackages_hardkernel_latest.bcc</li>
<li>linuxPackages_hardkernel_latest.bpftrace</li>
<li>linuxPackages_hardkernel_latest.perf</li>
<li>linuxPackages_hardkernel_latest.systemtap</li>
<li>linuxPackages_latest-libre.bcc</li>
<li>linuxPackages_latest-libre.bpftrace</li>
<li>linuxPackages_latest-libre.oci-seccomp-bpf-hook</li>
<li>linuxPackages_latest-libre.systemtap</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>linuxPackages_xanmod.perf: log was empty</li>
  </ul>
</details>
<details>

  <summary>4 packages failed to build and are new build failures:</summary>
  <ul>
<li>linuxPackages_xanmod.bcc: log was empty</li>
<li>linuxPackages_xanmod.bpftrace: log was empty</li>
<li>linuxPackages_xanmod.oci-seccomp-bpf-hook: log was empty</li>
<li>linuxPackages_xanmod.systemtap: log was empty</li>
  </ul>
</details>
<details>
<summary>
58 packages built:
</summary>
<ul>
<li>cargo-flamegraph</li>
<li>linuxPackages-libre.perf</li>
<li>linuxPackages.bcc (linuxPackages_5_10.bcc)</li>
<li>linuxPackages.bpftrace (linuxPackages_5_10.bpftrace)</li>
<li>linuxPackages.oci-seccomp-bpf-hook (linuxPackages_5_10.oci-seccomp-bpf-hook)</li>
<li>linuxPackages.perf (linuxPackages_5_10.perf ,linuxPackages_hardened.perf)</li>
<li>linuxPackages.systemtap (linuxPackages_5_10.systemtap)</li>
<li>linuxPackages_4_14.bcc</li>
<li>linuxPackages_4_14.bpftrace</li>
<li>linuxPackages_4_14.systemtap</li>
<li>linuxPackages_4_19.bcc</li>
<li>linuxPackages_4_19.bpftrace</li>
<li>linuxPackages_4_19.systemtap</li>
<li>linuxPackages_4_4.bcc</li>
<li>linuxPackages_4_4.bpftrace</li>
<li>linuxPackages_4_4.systemtap</li>
<li>linuxPackages_4_9.bcc</li>
<li>linuxPackages_4_9.bpftrace</li>
<li>linuxPackages_4_9.systemtap</li>
<li>linuxPackages_5_12.bcc</li>
<li>linuxPackages_5_12.bpftrace</li>
<li>linuxPackages_5_12.oci-seccomp-bpf-hook</li>
<li>linuxPackages_5_12.perf (linuxPackages_latest_hardened.perf)</li>
<li>linuxPackages_5_12.systemtap</li>
<li>linuxPackages_5_13.bcc (linuxPackages_latest.bcc)</li>
<li>linuxPackages_5_13.bpftrace (linuxPackages_latest.bpftrace)</li>
<li>linuxPackages_5_13.oci-seccomp-bpf-hook (linuxPackages_latest.oci-seccomp-bpf-hook)</li>
<li>linuxPackages_5_13.perf (linuxPackages_latest.perf)</li>
<li>linuxPackages_5_13.systemtap (linuxPackages_latest.systemtap)</li>
<li>linuxPackages_5_4.bcc</li>
<li>linuxPackages_5_4.bpftrace</li>
<li>linuxPackages_5_4.oci-seccomp-bpf-hook</li>
<li>linuxPackages_5_4.perf</li>
<li>linuxPackages_5_4.systemtap</li>
<li>linuxPackages_hardened.bcc</li>
<li>linuxPackages_hardened.bpftrace</li>
<li>linuxPackages_hardened.oci-seccomp-bpf-hook</li>
<li>linuxPackages_hardened.systemtap</li>
<li>linuxPackages_latest-libre.perf</li>
<li>linuxPackages_latest_hardened.bcc</li>
<li>linuxPackages_latest_hardened.bpftrace</li>
<li>linuxPackages_latest_hardened.oci-seccomp-bpf-hook</li>
<li>linuxPackages_latest_hardened.systemtap</li>
<li>linuxPackages_lqx.bcc</li>
<li>linuxPackages_lqx.bpftrace</li>
<li>linuxPackages_lqx.oci-seccomp-bpf-hook</li>
<li>linuxPackages_lqx.perf</li>
<li>linuxPackages_lqx.systemtap</li>
<li>linuxPackages_testing_bcachefs.bcc</li>
<li>linuxPackages_testing_bcachefs.bpftrace</li>
<li>linuxPackages_testing_bcachefs.oci-seccomp-bpf-hook</li>
<li>linuxPackages_testing_bcachefs.perf</li>
<li>linuxPackages_testing_bcachefs.systemtap</li>
<li>linuxPackages_zen.bcc</li>
<li>linuxPackages_zen.bpftrace</li>
<li>linuxPackages_zen.oci-seccomp-bpf-hook</li>
<li>linuxPackages_zen.perf</li>
<li>linuxPackages_zen.systemtap</li>
</ul>
</details>

**Pull request merged**